python_src/src/lamp_py/performance_manager/l0_gtfs_static_mod.py

```diff
@@ -170,6 +137,63 @@ def generate_scheduled_trunk_headways(
     process_logger.log_complete()
 
 
+def load_temp_table_headway_gen(
+    fk_timestamp: int, db_manager: DatabaseManager
+) -> None:
+    """
+    load static schedule data into temp table for generating scheduled headways
+
+    this was selected because executing an update as a CTE statement causes very poor performance
+    """
+    load_headways_gen = (
+        sa.select(
+            StaticStopTimes.pk_id,
+            StaticStopTimes.departure_time,
+            sa.func.coalesce(
+                StaticStops.parent_station, StaticStops.stop_id
+            ).label("parent_station"),
+            StaticTrips.service_id,
+            StaticTrips.direction_id,
+            StaticTrips.trunk_route_id,
+            StaticTrips.branch_route_id,
+        )
+        .select_from(StaticStopTimes)
+        .join(
+            StaticStops,
+            sa.and_(
+                StaticStops.stop_id == StaticStopTimes.stop_id,
+                StaticStops.timestamp == StaticStopTimes.timestamp,
+            ),
+        )
+        .join(
+            StaticTrips,
+            sa.and_(
+                StaticTrips.trip_id == StaticStopTimes.trip_id,
+                StaticTrips.timestamp == StaticStopTimes.timestamp,
+            ),
+        )
+        .where(StaticStopTimes.timestamp == fk_timestamp)
+    )
```

Comment:
Load temporary table for each schedule file with `INSERT FROM SELECT` SQL command.

Reply:
how do you decide which table goes in the initial `SELECT` and which ones are in `JOIN`?

Reply:
It doesn't really matter, in this instance I think of `static_stop_times` as the main table we're starting from because the main piece of data we're after is the `departure_time` values. So I select from `static_stop_times` and then `JOIN` the other tables. 

It all gets mashed together by the DB engine optimizer into whatever the most efficient version will eventually be, but logically this flow makes the most sense to me personally.